### PR TITLE
feat(tangle-dapp): Backend Integration for Service Info Card and Participants table

### LINKED ISSUE #2063: [TASK] Integrate back-end on InfoCard on Service Details page
### Description
Fetch data to display service type, threshold, key, start and end date. This must be done with `GraphQL` since **polkadot** `storage` does not store timestamps when a service is created

Image

### LINKED ISSUE #2066: [TASK] Integrate back-end on Participants Table on Service Details page

### PR DESCRIPTION
**Pull request by vutuanlinh2k2**
## Summary of changes

_Provide a detailed description of proposed changes._

- Add Service Details Context to use across all components of Service Details page
- Retrieve and display data for Service Info Card on Service Details page
- Retrieve and display data for Participants Table on Service Details page
- Note: Need to update the UI of Service Info Card because we cannot get the starting time of service at the moment

### Proposed area of change

_Put an `x` in the boxes that apply._

- [ ] `apps/bridge-dapp`
- [ ] `apps/hubble-stats`
- [ ] `apps/stats-dapp`
- [x] `apps/tangle-dapp`
- [ ] `apps/testnet-leaderboard`
- [ ] `apps/faucet`
- [ ] `apps/zk-explorer`
- [ ] `libs/webb-ui-components`

### Reference issue to close (if applicable)

_Specify any issues that can be closed from these changes (e.g. `Closes #233`)._

- Closes #2063 
- Closes #2066 

### Screen Recording

_If possible provide a screen recording of proposed change._

https://github.com/webb-tools/webb-dapp/assets/69841784/428cd18c-7f64-4238-bbe2-4954ab8629f4

#### Service has ended case
CleanShot 2024-05-02 at 15 21 41@2x

---

### Code Checklist

_Please be sure to add .stories documentation if any additions are made to `libs/webb-ui-components`._

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: apps/tangle-dapp/context/ServiceDetailsContext.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      } catch (error) {
+        notificationApi({
+          message: 'Failed to load service',
+          variant: 'error',
+        });
+      }
+    };
+
+    fetchServiceDetails();
+  }, [rpcEndpoint, serviceId, notificationApi]);
```
Comment (AtelyPham) [inline apps/tangle-dapp/context/ServiceDetailsContext.tsx:66]: Why don't you utilize the usePolkadotApi hook here?

File: apps/tangle-dapp/data/serviceDetails/useServiceParticipants.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            ...(accountInfo ?? {}),
+          } satisfies ServiceParticipant;
+        })
+      );
+      setParticipants(participantsInfo);
+      setIsLoadingParticipants(false);
+    };
+
+    fetchParticipantsInfo();
+  }, [rpcEndpoint, serviceDetails?.participants]);
```
Comment (AtelyPham) [inline apps/tangle-dapp/data/serviceDetails/useServiceParticipants.ts:34]: Same here, consider using `usePolkadotApi`.

File: apps/tangle-dapp/data/validatorDetails/useValidatorInfoCard.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        setNominations(null);
+        setIsActive(false);
+      };
+
+      await Promise.all([fetchNameAndSocials(), fetchNominations()]);
+      setIsLoading(false);
+    };
+
+    fetchData();
+  }, [validatorAddress, rpcEndpoint, currentEra]);
```
Comment (AtelyPham) [inline apps/tangle-dapp/data/validatorDetails/useValidatorInfoCard.ts:84]: Same herể.

File: apps/tangle-dapp/data/validatorDetails/useValidatorInfoCard.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  getTotalRestakedFromRestakeRoleLedger,
+} from '../../utils/polkadot/restake';
+
+export default function useValidatorInfoCard(
+  rpcEndpoint: string,
+  validatorAddress: string
+) {
+  const { data: currentEra } = useCurrentEra();
+  const { data: ledgerOpt, isLoading: isLoadingLedgerOpt } =
+    useRestakingRoleLedger(validatorAddress);
```
Comment (AtelyPham) [inline apps/tangle-dapp/data/validatorDetails/useValidatorInfoCard.ts:19]: Think of creating a `RestakeContext` to share these props between the service and the restake page.

**Review by AtelyPham — COMMENTED** [review]
Left a few comments.

File: apps/tangle-dapp/data/validatorDetails/useValidatorInfoCard.ts (hunk 4 above)
Comment (vutuanlinh2k2) [reply apps/tangle-dapp/data/validatorDetails/useValidatorInfoCard.ts:19]: The `RestakeContext` is for handling the data of the current user, while the service page displays the data of the whole network. I think we can keep it like this for now

File: apps/tangle-dapp/data/serviceDetails/useServiceParticipants.ts (hunk 2 above)
Comment (vutuanlinh2k2) [reply apps/tangle-dapp/data/serviceDetails/useServiceParticipants.ts:34]: This does not use the api from PolkadotJS directly

**Pull request merged**